src/NewScout/screens/ArticleDetailsScreen.js

```diff
@@ -165,139 +165,165 @@ class SuggestionsScreen extends React.Component {
 }
 
 class ArticleDetailsView extends React.Component {
-    constructor(props) {
-      super(props);
-      this.state = {
-        data: ['Test 1', 'Test 2', 'Test 3'],
-        showSuggestions: false,
-      };
-      this.callAPI(this.props.route.params.articleSlug);
-    }
-  
-    callAPI = (articleSlug) => {
-      return fetch(`http://www.newscout.in/api/v1/articles/${articleSlug}/?domain=newscout&format=json`)
-        .then(response => response.json())
-        .then(json => {
-          this.setState({
-            data: [json.body.article],
-          });
-        })
-        .catch(error => {
-          console.error(error);
-        });
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: [], //'Test 1', 'Test 2', 'Test 3'
+      showSuggestions: false,
+      loading: false,
     };
+    this.callAPI(this.props.route.params.articleSlug);
+  }
 
-    loadItems() {
-      this.setState({
-        data: [
-          'Test 1',
-          'Test 2',
-          'Test 3',
-          'Test 4',
-          'Test 5',
-          'Test 6',
-          'Test 7',
-        ],
+  callAPI = (articleSlug) => {
+    this.setState({
+      loading: true,
+    });
+    return fetch(
+      `http://www.newscout.in/api/v1/articles/${articleSlug}/?domain=newscout&format=json`,
+    )
+      .then((response) => response.json())
+      .then((json) => {
+        this.setState({
+          data: [json.body.article],
+          loading: false,
+        });
+      })
+      .catch((error) => {
+        console.error(error);
+        this.setState({
+          loading: false,
```

Comment:
@gmcodebuster if there is error, we need to load some sort of "Something went wrong" screen and terminate loading. What do you think?